pkg/azurefile/controllerserver.go

```diff
@@ -229,3 +276,34 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
 func (d *Driver) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
 	return nil, status.Error(codes.Unimplemented, "ControllerExpandVolume is not yet implemented")
 }
+
+// getShareUrl: sourceVolumeID is the id of source file share, returns a ShareURL of source file share.
+// A ShareURL < https://<account>.file.core.windows.net/<fileShareName> > represents a URL to the Azure Storage share allowing you to manipulate its directories and files.
+// e.g. The ID of source file share is #fb8fff227be6511e9b24123#createsnapshot-volume-1. Returns https://fb8fff227be6511e9b24123.file.core.windows.net/createsnapshot-volume-1
+func (d *Driver) getShareUrl(sourceVolumeID string) (azfile.ShareURL, error) {
+	resourceGroupName, accountName, fileShareName, err := getFileShareInfo(sourceVolumeID)
+	if err != nil {
+		klog.Errorf("getFileShareInfo(%s) failed with error: %v", sourceVolumeID, err)
+		return azfile.ShareURL{}, err
+	}
+
+	if resourceGroupName == "" {
+		resourceGroupName = d.cloud.ResourceGroup
+	}
+
+	accountKey, err := d.cloud.GetStorageAccesskey(accountName, resourceGroupName)
+	if err != nil {
+		return azfile.ShareURL{}, fmt.Errorf("no key for storage account(%s) under resource group(%s), err %v", accountName, resourceGroupName, err)
+	}
+
+	credential, err := azfile.NewSharedKeyCredential(accountName, accountKey)
+	if err != nil {
+		klog.Errorf("NewSharedKeyCredential(%s) in CreateSnapshot failed with error: %v", accountName, err)
+		return azfile.ShareURL{}, err
+	}
+
+	u, _ := url.Parse(fmt.Sprintf(fileUrlTemplate, accountName, d.cloud.Environment.StorageEndpointSuffix))
```

Comment:
we may still check error here, otherwise u may be nil

Reply:
also check u is not nil

Reply:
done.